### DISCUSSION TITLE
SSO using twitter

### DISCUSSION THREAD

**Discussion by alihassan161820** [directus/directus#10186] [Archive]
Can you provide correct configuration for twitter sso 

```
AUTH_TWITTER_DRIVER="oauth2"
AUTH_TWITTER_CLIENT_ID="**"
AUTH_TWITTER_CLIENT_SECRET="**"
AUTH_TWITTER_AUTHORIZE_URL="https://twitter.com/i/oauth2/authorize"
AUTH_TWITTER_ACCESS_URL="https://api.twitter.com/oauth/access_token"
AUTH_TWITTER_ALLOW_PUBLIC_REGISTRATION="true"
AUTH_TWITTER_SCOPE="users.read"
```

according twitter docs oauth2 is still beta version so what's the solution for login using twitter ?

**Comment by rijkvanzanten** [comment]
I believe you have to setup an _app_ on the Twitter developer portal to retrieve those IDs and Secrets (https://developer.twitter.com/en/apps)

**Reply by aidenfoxx** [reply]
And needless to say, you will have to use the twitter OAuth 2.0 beta since there is no support for OAuth 1.0 in Directus.

**Reply by alihassan161820** [reply]
@rijkvanzanten i already did but i removed them to post a discussions 

@aidenfoxx Oauth 2.0 beta is still not available for every one to  use

**Reply by rijkvanzanten** [reply]
@alihassan161820 ah that wasn't clear from the message above. Wrapping the vars in triple backticks reveals that you added `**` as a placeholder.

**Reply by aidenfoxx** [reply]
Looking into the docs for the beta, I don't believe their OAuth 2 implementation would work anyway since doesn't provide a standard userinfo endpoint. We'd have to add a special exception for Twitter to extract the non-standard `user_id` variable from their callback request and send that to the `/2/users/by` endpoint. 🤔
